api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -344,6 +344,24 @@ default void invalidateTable(TableIdentifier identifier) {}
    * @throws AlreadyExistsException if the table already exists in the catalog.
    */
   default Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
+    return registerTable(
+        identifier, metadataFileLocation, false /* register only if it does not exist */);
+  }
+
+  /**
+   * Register a table with the catalog, optionally overwrite existing table metadata.
+   *
+   * <p><strong>Note:</strong> Overwriting an existing table may result in a new table UUID.
```

Comment:
Does this mean we won't use the UUID from the metadata?

Reply:
Actually, let me reword this. What I meant is that the existing table UUID might change after the register with overwrite (original UUID might change to UUID from the metadata.json if they are different)